rules_python_zip/simplepack.go

```diff
@@ -501,6 +502,14 @@ if need_unzip and isinstance(__loader__, zipimport.zipimporter):
     tempdir_create_pid = os.getpid()
     atexit.register(clean_tempdir_parent_only, tempdir)
     sys.path.insert(0, tempdir)
+    # Handle linux signal terminate by calling exit, so atexit code executes.
+    old_handler = None
+    def sig_exit(*args):
+        if sys.path[0].endswith('_pyzip'):
+            shutil.rmtree(sys.path[0])
+        if old_handler:
+            old_handler(*args)
+    old_handler = signal.signal(signal.SIGTERM, sig_exit)
```

Comment:
I thought we decided having pyz code provide signal handlers was a bad idea, given that it's a reasonably common thing for code that might get packaged into a pyz binary to want to do.

Reply:
That was the plan, but while it works properly for the lightning forecast (what I was testing it on originally) it failed to work properly for any of the multi-threaded tools I've got. Had to test a bunch of permutations to figure out how to get it working correctly.

Reply:
Hmm, could you elaborate? Why did it work in some cases without modifying pyz, but not in others? In what way did it not work? Could it be this forking issue? Are you using multiprocessing? The code here LGTM, but I'd favor doing signal handling elsewhere unless this is the only way (which seems unlikely...).

Reply:
(Or happy to chat in person)